Drv/LinuxGpioDriver/LinuxGpioDriver.cpp

```diff
@@ -0,0 +1,314 @@
+    if (return_value != 0) {
+        status = errno_to_file_status(errno);
+        this->log_WARNING_HI_OpenChipError(Fw::String(device), Os::FileStatus(static_cast<Os::FileStatus::T>(status)));
+        return status;
+    }
+    this->log_DIAGNOSTIC_OpenChip(Fw::String(chip_info.name), Fw::String(chip_info.label), chip_info.lines);
+    // Check if the GPIO line exists
+    if (gpio >= chip_info.lines) {
+        this->log_WARNING_HI_OpenPinError(Fw::String(device), gpio,
+                                          Os::FileStatus(static_cast<Os::FileStatus::T>(status)));
+        return status;
+    }
+    // Set up pin and set file descriptor for it
+    PlatformIntType pin_fd = -1;
+    switch (configuration) {
+        // Cascade intended
+        case GPIO_OUTPUT:
+        case GPIO_INPUT:
+            status = this->setupLineHandle(chip_descriptor, gpio, configuration, default_state, pin_fd);
+            break;
+        // Cascade intended
+        case GPIO_INTERRUPT_RISING_EDGE:
+        case GPIO_INTERRUPT_FALLING_EDGE:
+        case GPIO_INTERRUPT_BOTH_RISING_AND_FALLING_EDGES:
+            status = this->setupLineEvent(chip_descriptor, gpio, configuration, pin_fd);
+            break;
+        default:
+            FW_ASSERT(0);
+            break;
+    }
+    // Final status check
+    if (status != Os::File::Status::OP_OK) {
+        this->log_WARNING_HI_OpenPinError(Fw::String(device), gpio,
+                                          Os::FileStatus(static_cast<Os::FileStatus::T>(status)));
+    } else {
+        this->m_fd = pin_fd;
+        this->m_configuration = configuration;
+    }
+    return status;
+}
+
+Drv::GpioStatus LinuxGpioDriver ::gpioRead_handler(const NATIVE_INT_TYPE portNum, Fw::Logic& state) {
+    Drv::GpioStatus status = Drv::GpioStatus::INVALID_MODE;
+    if (this->m_configuration == GpioConfiguration::GPIO_INPUT) {
+        struct gpiohandle_data values;
+        ::memset(&values, 0, sizeof values);
+        PlatformIntType return_value = ioctl(this->m_fd, GPIOHANDLE_GET_LINE_VALUES_IOCTL, values);
+        if (return_value != 0) {
+            status = errno_to_gpio_status(errno);
+        } else {
+            state = values.values[0] ? Fw::Logic::HIGH : Fw::Logic::LOW;
+        }
+    }
+    return status;
+}
+
+Drv::GpioStatus LinuxGpioDriver ::gpioWrite_handler(const NATIVE_INT_TYPE portNum, const Fw::Logic& state) {
+    Drv::GpioStatus status = Drv::GpioStatus::INVALID_MODE;
+    if (this->m_configuration == GpioConfiguration::GPIO_OUTPUT) {
+        struct gpiohandle_data values;
+        ::memset(&values, 0, sizeof values);
+        values.values[0] = (state == Fw::Logic::HIGH) ? 1 : 0;
+        PlatformIntType return_value = ioctl(this->m_fd, GPIOHANDLE_SET_LINE_VALUES_IOCTL, values);
+        if (return_value != 0) {
+            status = errno_to_gpio_status(errno);
+        }
+    }
+    return status;
```

Comment:
The status returned here will never be OP_OK

Reply:
Extremely good catch!